### ISSUE TITLE
[BUG] Incorrect behavior of read_parquet() on list of files - generates unusable DataFrame

### ISSUE THREAD

**Issue by mzanotto** [rapidsai/cudf#11096]
**Describe the bug**
Running some tests I bumped into a serious issue.
When passing an array of files to read_parquet(), the generated DataFrame is incorrect.

What seems to happen is that the data is concatenated correctly, but the accessible rows are capped to the rows of the last file in the list.
If you run things like .unique() on a column, .describe() or .info() you can see all the data is there. Despite that, you can't access the "invisible" portion of the DataFrame nor run things like set_index().

**Steps/Code to reproduce bug**
```

import cudf

# Creating two dummy datasets and saving to parquet
df_1_rows = 100
df_1 = cudf.DataFrame({"id":range(df_1_rows),"a":[1]*df_1_rows})
df_1.to_parquet("./cudf_test/df_1.parquet")

df_2_rows = 2000
df_2 = cudf.DataFrame({"id":range(200,200+df_2_rows),"a":[2]*df_2_rows})
df_2.to_parquet("./cudf_test/df_2.parquet")

# Loading the generated parquet files in two different orders
test_files_order_1 = ["./cudf_test/df_1.parquet", "./cudf_test/df_2.parquet"]
df_o1 = cudf.read_parquet(test_files_order_1)

test_files_order_2 = ["./cudf_test/df_2.parquet", "./cudf_test/df_1.parquet"]
df_o2 = cudf.read_parquet(test_files_order_2)

print("Comparing lengths: {} vs {}".format(len(df_o1), len(df_o2)))
```
Which returns
`Comparing lengths: 2000 vs 100`

Then
```
df_o1.info(memory_usage=True)

<class 'cudf.core.dataframe.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   id      2100 non-null   int64
 1   a       2100 non-null   int64
dtypes: int64(2)
memory usage: 32.8 KB
```
and

```
df_o2.info(memory_usage=True)

<class 'cudf.core.dataframe.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   id      2100 non-null   int64
 1   a       2100 non-null   int64
dtypes: int64(2)
memory usage: 32.8 KB
```

Showing that the data itself is the same but the RangeIndex's are different and incompatible with the Non-Null counts in both cases.

**Expected behavior**
The DataFrame created from `read_parquet()` on list of files should have all the read data visible and available to use.

**Environment overview (please complete the following information)**
 - Environment location: AWS g5.xlarge, NVIDIA Deep Learning AMI (old version) https://aws.amazon.com/marketplace/pp/prodview-e7zxdqduz4cbs
 - Method of cuDF install: Docker (image derived from nvcr.io/nvidia/merlin/merlin-pytorch-training:22.03)

**Environment details**
I can't find the `cudf/print_env.sh` script on the Docker image but I'm using the above AMI and Docker image without changes on the NVIDIA tooling side.
Relevant details:
cudf version: 22.02.00a+309.gdad51a548e (but was able to replicate on 22.04 after manual update)
CUDA version: 11.6
Driver version: 460.73.01

**Comment by mzanotto** [comment]
Quick update: the problem seems to be in the index.

If you do something like

`cudf.DataFrame.from_arrow(df_o1.to_arrow(preserve_index=False))`

or

`cudf.DataFrame({"id":cudf.from_dlpack(df_o1["id"].to_dlpack()),"a":cudf.from_dlpack(df_o1["a"].to_dlpack())})`

the obtained DataFrames seem to be OK.

Unfortunately using `.reset_index(drop=True)` or `.reindex(index=range(...))` don't seem to work.

**Comment by galipremsagar** [comment]
> @galipremsagar am I reading the PR correctly in that you think this only happens for DataFrames using `RangeIndex`?

Yes

> I see the fix is targeted to **v22.08**. Is setting an index manually before saving the parquet files a valid temporary fix for this? I tried it in the example I provided in the bug report and it seems to be working in that specific case.


Yes, when you do `to_parquet(index=True)` the `RangeIndex` is materialized and saved as an `IntegerIndex`, which will be retrieved as an `IntegerIndex`.
